### DISCUSSION TITLE
Canadian Support

### DISCUSSION THREAD

**Discussion by Cmaxster** [pwillworth/dfkreport#8] [Ideas]
Hello! Thank you so much for putting this project together. So far I find it's very helpful and a bit more accurate than dfktracker.app.

I realize that you are based out of the U.S., and the majority of NA players are probably American, but I and my fellow hosers would be eternally grateful if you could find a way to translate these numbers to Canadian dollars and our "Adjusted Cost Base" method required by our tax system.

If that's too much to take on, at the very least it would be fantastic to just have the CAD option to quickly convert from the default USD amounts.

Thanks again!

**Comment by Cmaxster** [comment]
In case you're up for the challenge, this is a quick primer on how our system works:
https://www.adjustedcostbase.ca/blog/how-to-calculate-adjusted-cost-base-acb-and-capital-gains/

**Reply by pwillworth** [reply]
Interesting, thanks for giving me a heads up about this Adjusted Cost Basis, I was not aware of it and will plan on adding it.  Was definitely planning to add the option to display in other fiat currencies, so that will be coming.  Seems like it might be more important to make the ACB option available first though, would you agree?

**Reply by Cmaxster** [reply]
Yeah, I don't think it's too drastically different from the American method, so it probably wouldn't be too tricky to add in. Also adding in CAD would just be a simple currency conversion, you can probably pull the rate easily from an API. You could probably have a drop-down and just pull values for whatever currencies seem appropriate to include.

I've been combing through the Bounty submissions to try to find a viable Canadian option and most of them seem to have omitted it. DFKTracker.app has CAD listed, but the numbers seem way too high for some reason, and their CSV files only output in USD at this point.

Speaking for myself, I would love to see the ACB option added in first. Speaking from a broader sense, I'm not sure what the demographics of the game are right now, but I suspect Canada and Europe would probably be the next largest English speaking user base, so it would certainly be helpful to many of us I think.

I dabble a bit in development myself, but am a bit less experienced. I was starting to draw up plans to build out an app to pull data and calculate ACB but I haven't gotten very far. I'm also not that experienced with Accounting or Tax rules, and crypto rules are still hazy, so I'm still researching how it works. I think it boils down to splitting transactions into Business or Investment Income, and Capital Gains which is pretty much what you already have I think. Let me know how it goes for you if you're able to take it on and add it in.

I found this: https://github.com/EmilMaric/cad-capital-gains

**Reply by pwillworth** [reply]
Sounds good, thanks for sharing that repo, might come in handy.  Yeah, doing the multiple currencies thing is a little more involved to do it right, and easier to adjust for manually than the ACB, so makes sense to tackle ACB first.  I'll update this thread when any of these updates we've talked about get published.

**Reply by pwillworth** [reply]
Hi @Cmaxster , I just published an initial version of the Adjusted Cost Basis option to Lila's Ledger.  Wanted to give you a heads up on that incase you want to try it out, and also ask if you have any thoughts on superficial capital loss rules and what would be helpful from your perspective.  My initial thoughts were that with the constant swapping of different tokens that most people do, applying the superficial capital loss rule would pretty much negate any loss on a trade and doesn't really seem to be in the spirit of the rule either.

**Reply by Cmaxster** [reply]
Hi @pwillworth! Thanks for the heads up, I'll definitely check it out. Do you mind if we chat on Discord? I couldn't find a DM feature on Github so I'll just post my username here. I'm Cmaxster#5372. Send me a DM when you get a chance.
